kernel/vconv.ml

```diff
@@ -90,28 +95,39 @@ and conv_atom env pb k a1 stk1 a2 stk2 cu =
 (*  Pp.(msg_debug (str "conv_atom(" ++ pr_atom a1 ++ str ", " ++ pr_atom a2 ++ str ")")) ; *)
   match a1, a2 with
   | Aind ((mi,_i) as ind1) , Aind ind2 ->
-    if Ind.CanOrd.equal ind1 ind2 && compare_stack stk1 stk2 then
+    if Names.Ind.CanOrd.equal ind1 ind2 && compare_stack stk1 stk2 then
       let ulen = Univ.AbstractContext.size (Environ.mind_context env mi) in
       if ulen = 0 then
         conv_stack env k stk1 stk2 cu
       else
         match stk1 , stk2 with
         | Zapp args1 :: stk1' , Zapp args2 :: stk2' ->
-          assert (ulen <= nargs args1);
-          assert (ulen <= nargs args2);
-          let u1 = Array.init ulen (fun i -> uni_lvl_val (arg args1 i)) in
-          let u2 = Array.init ulen (fun i -> uni_lvl_val (arg args2 i)) in
-          let u1 = Univ.Instance.of_array u1 in
-          let u2 = Univ.Instance.of_array u2 in
+          assert (0 < nargs args1);
+          assert (0 < nargs args2);
+          let u1 = uni_instance (arg args1 0) in
+          let u2 = uni_instance (arg args2 0) in
           let cu = convert_instances ~flex:false u1 u2 cu in
-          conv_arguments env ~from:ulen k args1 args2
+          conv_arguments env ~from:1 k args1 args2
             (conv_stack env k stk1' stk2' cu)
         | _, _ -> assert false (* Should not happen if problem is well typed *)
     else raise NotConvertible
   | Aid ik1, Aid ik2 ->
     if Vmvalues.eq_id_key ik1 ik2 && compare_stack stk1 stk2 then
+      let ulen = table_key_instance env ik1 in
+      if ulen = 0 then
         conv_stack env k stk1 stk2 cu
-      else raise NotConvertible
+      else
+        match stk1 , stk2 with
+        | Zapp args1 :: stk1' , Zapp args2 :: stk2' ->
+          assert (0 < nargs args1);
+          assert (0 < nargs args2);
+          let u1 = uni_instance (arg args1 0) in
+          let u2 = uni_instance (arg args2 0) in
+          let cu = convert_instances ~flex:true u1 u2 cu in
```

Comment:
```suggestion
          let cu = convert_instances ~flex:false u1 u2 cu in
```
flex:true is when the constant is transparent

Reply:
Then the doc in conversion.ml is incorrect. It states
```
(* [flex] should be true for constants, false for inductive types and
   constructors. *)
```
but then this should read "true for transparent constants".

Reply:
yes

Reply:
(NB I think vconv is only called with checked_universes so it doesn't really matter here)